File: python-bindings/tests/yaml_source_values_conformance.py
"""Read generic YAML source values through public Python exports."""


def observe_yaml_sources(fixture: dict) -> dict:
    """Preserve the difference between shared game data and per-game local paths."""
    import classic_config

    sources = [getattr(classic_config.YamlSource, name) for name in fixture["sources"]]
    rows = []
    for name, source in zip(fixture["sources"], sources, strict=True):
        if repr(source) != "YamlSource." + name or str(source) != name:
            raise ValueError("source representation lost its variant identity")
        rows.append(
            {
                "id": name,
                "path": source.path(fixture["game"]).replace("\\", "/"),
                "name": source.display_name(),
                "gameName": source.display_name_with_game(fixture["game"]),
            }
        )
    return {
        "sources": rows,
        "equal": sources[0] == classic_config.YamlSource.MAIN,
        "different": sources[0] != sources[1],
        "distinct": len(set(sources)),
    }

File: node-bindings/classic-node/__test__/yaml_source_values_conformance.ts
import * as classic from "../index.js";

/** Read generic source paths, names and enum distinctions from public exports. */
export function observeYamlSources(fixture: any): Record<string, unknown> {
    const variants: Record<string, classic.JsYamlSource> = {
        MAIN: classic.JsYamlSource.Main,
        IGNORE: classic.JsYamlSource.Ignore,
        GAME: classic.JsYamlSource.Game,
        GAME_LOCAL: classic.JsYamlSource.GameLocal,
        TEST: classic.JsYamlSource.Test
    };
    const sources = fixture.sources.map((name: string) => {
        if (!(name in variants)) throw new Error("unsupported source variant");
        return variants[name];
    });
    return {
        sources: sources.map((source: classic.JsYamlSource, index: number) => ({
            id: fixture.sources[index],
            path: classic.getYamlSourcePath(source, fixture.game).replaceAll("\\", "/"),
            name: classic.getYamlSourceDisplayName(source),
            gameName: classic.getYamlSourceDisplayNameWithGame(source, fixture.game)
        })),
        equal: sources[0] === classic.JsYamlSource.Main,
        different: sources[0] !== sources[1],
        distinct: new Set(sources).size
    };
}
